### PR TITLE
Windows keybase.exe launch/retry

### PR DESCRIPTION
**Pull request by zanderz**
If we can't connect, we'll try to launch the service(s).
Testing with build https://ci.keyba.se/job/windows-installer/783/

### CODE REVIEW

**Comment by chrisnojima** [comment]
lets not request reviews from individuals. just tag @keybase/react-hackers so we can all see it

File: shared/desktop/renderer/index.js [hunk 1]
```diff
@@ -93,6 +94,15 @@ function setupApp(store) {
     })
   })
 
+  if (os.platform() === 'win32') {
+    // After a delay, see if we're connected, and try starting keybase if not
+    setTimeout(() => {
+      eng.listenOnNotConnected('win-service-check', () => {
```
Comment (chrisnojima) [inline shared/desktop/renderer/index.js]: don't think this is a thing

File: shared/desktop/renderer/index.js [hunk 2]
```diff
@@ -58,7 +59,7 @@ function setupAvatar() {
 function setupApp(store) {
   setupSource()
   disableDragDrop()
-  makeEngine()
+  var eng = makeEngine()
```
Comment (chrisnojima) [inline shared/desktop/renderer/index.js]: could be a const

File: shared/desktop/app/installer.js [hunk 3]
```diff
@@ -1,9 +1,11 @@
 // @flow
 import {app} from 'electron'
 import exec from './exec'
+
```
Comment (chrisnojima) [inline shared/desktop/app/installer.js]: 🔥

File: shared/desktop/app/installer.js [hunk 4]
```diff
@@ -1,9 +1,11 @@
...
 import {appInstallerPath, appBundlePath} from './paths'
 import {quit} from './ctl'
 import {runMode} from '../../constants/platform.desktop'
+import os from 'os'
```
Comment (chrisnojima) [inline shared/desktop/app/installer.js]: dont need this, we have isWindows in platform.desktop

File: shared/desktop/app/index.js [hunk 5]
```diff
@@ -83,6 +84,11 @@ function start() {
     })
   })
 
+  ipcMain.on('win-service-check', (event, arg) => {
```
Comment (chrisnojima) [inline shared/desktop/app/index.js]: should wrap this in a isWindows check also

Comment (chrisnojima) [inline shared/desktop/app/index.js]: or this should be more generic like 'startService' and on non-windows systems it effectively does nothing, that might be better

File: shared/desktop/renderer/index.js (hunk 1 above)
Comment (zanderz) [inline shared/desktop/renderer/index.js]: sorry about that, lost some changes in a rebase somehow, stand by...

**Comment by zanderz** [comment]
@keybase/react-hackers

File: shared/desktop/app/ctl.js [hunk 6]
```diff
@@ -6,6 +6,8 @@ import exec from './exec'
 export function ctlStop(callback: any) {
   const binPath = keybaseBinPath()
   const args = ['ctl', 'stop', '--exclude=app']
+  // TODO: on Win32, we would need: args = ['ctl', 'stop', '--kill-kbfs']
```
Comment (gabriel) [inline shared/desktop/app/ctl.js]: Is `kill-kbfs` really an option to ctl stop?

File: shared/desktop/app/index.js [hunk 7]
```diff
@@ -83,6 +85,13 @@ function start() {
     })
   })
 
+  ipcMain.on('kb-service-check', (event, arg) => {
+    if (isWindows) {
+      console.log('kb-service-check')
```
Comment (gabriel) [inline shared/desktop/app/index.js]: This log statement could be better?

File: shared/desktop/app/paths.js [hunk 8]
```diff
@@ -38,6 +38,12 @@ export function appInstallerPath() {
 
 // Path to keybase executable (darwin only), null if not available
 export function keybaseBinPath() {
+  if (os.platform() === 'win32') {
+      var kbPath = app.getPath('appData').replace('Roaming', 'Local')
+      if (kbPath === null) kbPath = process.env.LOCALAPPDATA
+      if (kbPath === null) return null
```
Comment (gabriel) [inline shared/desktop/app/paths.js]: Might be useful to log this case as warning

File: shared/desktop/app/ctl.js (hunk 6 above)
Comment (zanderz) [inline shared/desktop/app/ctl.js]: On Windows it is

**Pull request merged**